pmd-javascript/src/test/resources/net/sourceforge/pmd/lang/ecmascript/rule/codestyle/xml/UnnecessaryBlock.xml

```diff
@@ -192,4 +201,44 @@ try {
 }
         ]]></code>
     </test-code>
+
+    <test-code>
+        <description>Ok, destructure assigments</description>
```

Comment:
Note, that this sample code doesn't produce a UnnecessaryBlock, even with your rule improvements.

The only way to reproduce the bug is by not using semicolons. A simple example is

```javascript
import { foo } from './someScript.js' // missing semicolon
const { isLoggedIn } = useAuth();  // <--- violation here: unnecessary block
```

(I've updated #2305 meanwhile)

Please add a separate test case for that.

Reply:
`[not(preceding::EmptyStatement)]` fixes not using semicolons case, but it not violates in these cases "Bad, for", "Bad, for in", "Bad, if", "Bad, try".
```
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   "Bad, for" failed
[ERROR] ruleTests().Bad, for in
[INFO]   Run 1: PASS
[ERROR]   Run 2: "Bad, for in" failed
[INFO] 
[ERROR]   "Bad, if" failed
[ERROR]   "Bad, try" failed
[INFO] 
[ERROR] Tests run: 154, Failures: 0, Errors: 4, Skipped: 0
```

Reply:
fixed by this rule:
```xpath
                    //Scope[
                        count(ancestor::*) = 1
                        and not(preceding::EmptyStatement)
                    ]
                    | //SwitchCase[./Scope]
                    | //(Scope|Block)[
                        ./(Scope|Block)
                        and count(./*) = 1
                    ]
```